vms/platformvm/state/state.go

```diff
@@ -808,6 +837,42 @@ func (s *state) SetSubnetOwner(subnetID ids.ID, owner fx.Owner) {
 	s.subnetOwners[subnetID] = owner
 }
 
+func (s *state) GetSubnetManager(subnetID ids.ID) (ids.ID, []byte, error) {
+	if chainIDAndAddr, exists := s.subnetManagers[subnetID]; exists {
+		return chainIDAndAddr.ChainID, chainIDAndAddr.Addr, nil
+	}
+
+	if chainIDAndAddr, cached := s.subnetManagerCache.Get(subnetID); cached {
+		return chainIDAndAddr.ChainID, chainIDAndAddr.Addr, nil
+	}
+
+	chainIDAndAddrBytes, err := s.subnetManagerDB.Get(subnetID[:])
+	if err == database.ErrNotFound {
+		s.SetSubnetManager(subnetID, ids.Empty, []byte{})
```

Comment:
Why are we calling this here?

Reply:
My thought was to cache the `{ids.Empty, []byte{}}` as the subnet manager when we first do `GetSubnetManager` on a newly created Subnet

Reply:
This writes to the database though - no?